Apps/AccountDataAccess/src/Patient/PatientRepository.cs

```diff
@@ -80,55 +90,113 @@ public async Task<PatientQueryResult> Query(PatientQuery query, CancellationToke
             };
         }
 
-        private async Task<ApiResult<PatientModel>> GetDemographicsAsync(OidType type, string identifier, PatientIdentifierType identifierType, bool disableIdValidation = false)
+        private static bool ShouldCheckCache(PatientDetailSource source)
         {
-            ApiResult<PatientModel> apiResult = new()
+            return source switch
             {
-                ResourcePayload = this.GetFromCache(identifier, identifierType),
+                PatientDetailSource.AllCache => true,
+                PatientDetailSource.EmpiCache => true,
+                PatientDetailSource.PhsaCache => true,
+                _ => false,
             };
+        }
 
-            if (apiResult.ResourcePayload == null)
+        private static bool ShouldCheckPhsa(PatientDetailSource source)
+        {
+            return source switch
             {
-                if (identifierType == PatientIdentifierType.Phn && !PhnValidator.IsValid(identifier))
-                {
-                    this.logger.LogDebug("The PHN provided is invalid");
-                    throw new ProblemDetailsException(ExceptionUtility.CreateProblemDetails(ErrorMessages.PhnInvalid, HttpStatusCode.NotFound, nameof(PatientRepository)));
-                }
+                PatientDetailSource.All => true,
+                PatientDetailSource.AllCache => true,
+                PatientDetailSource.Phsa => true,
+                PatientDetailSource.PhsaCache => true,
+                _ => false,
+            };
+        }
 
-                apiResult = await this.clientRegistriesDelegate.GetDemographicsAsync(type, identifier, disableIdValidation).ConfigureAwait(true);
+        private static bool ShouldCheckEmpi(PatientDetailSource source)
+        {
+            return source switch
+            {
+                PatientDetailSource.All => true,
+                PatientDetailSource.AllCache => true,
+                PatientDetailSource.Empi => true,
+                PatientDetailSource.EmpiCache => true,
+                _ => false,
+            };
+        }
```

Comment:
Nit: these could probably be handled more elegantly if the PatientDetailSource enum used a \[Flags\] attribute and had 3 members: Empi, Phsa, Cache.  Then you could determine if one of them applied by calling `source.HasFlag()`.

Reply:
I've thought about that too 😄.. let me rethink if it worth changing - it is not 'ideal' in a service interface contract as it is very implementation specific, however, it might be justified..

Reply:
@MikeLyttle  Discussed with @ytqsl this will be addressed in a new story.